code/ship/ship.cpp

```diff
@@ -7419,10 +7419,20 @@ void ship_render_show_ship_cockpit(object *objp)
 	model_render_params render_info;
 
 	render_info.set_object_number(OBJ_INDEX(objp));
-	render_info.set_replacement_textures(Ships[objp->instance].ship_replacement_textures);
+
+	// update any replacement and/or team color textures (wookieejedi), then render
+	ship* shipp = &Ships[objp->instance];
+	ship_info* sip = &Ship_info[Ships[objp->instance].ship_info_index];
+
+	render_info.set_replacement_textures(shipp->ship_replacement_textures);
```

Comment:
There's no need to check that the textures are valid before calling? Does `set_replacement_textures` already check?

Reply:
Good question. That function does not have that `is_team_color_set` check within it. As such, all other calls of `render_info.set_replacement_textures` had that same `if` check.